src/tutorials/boss_tutorial.md

```diff
@@ -0,0 +1,4330 @@
+    function activities.shoot.start(activity, deltaTime)
+        local tmpTarget = GetRandomTarget()
+        BOSS_GEO:LookAtContinuous(tmpTarget, false, .5)
+        Task.Wait(1)
+        target = tmpTarget
+        tmpTarget = nil
+    end
+
+    -- If a player enters the shoot perimeter of the AI, add
+    -- them to the shootTargets table
+    function OnPlayerEnterShoot(trigger, player)
+        if Object.IsValid(player) and player:IsA("Player") then
+            table.insert(shootTargets, player)
+        end
+    end
+
+    -- If the player leaves the shoot perimeter of the AI, remove
+    -- them from the shootTargets table.
+    function OnPlayerLeftShoot(trigger, player)
+        if Object.IsValid(player) and player:IsA("Player") then
+            for index, value in ipairs(shootTargets) do
+                if value == player then
+                    table.remove(shootTargets, index)
+
+                    if(target == player and Object.IsValid(BOSS_GEO)) then
+                        BOSS_GEO:StopRotate()
+                    end
+
+                    break
+                end
+            end
+        end
+    end
+
+    -- When the AI receives damage, broadcast that to the player
+    -- who caused the damage.
+    local function OnDamaged(obj, damage)
+        if not sentUpdateBossHealthBar then
+            Events.BroadcastToAllPlayers("CanUpdateBossHealthBar")
+            sentUpdateBossHealthBar = true
+        end
+    end
+
+    DAMAGEABLE.damagedEvent:Connect(OnDamaged)
+
+    SHOOT_PERIMETER.beginOverlapEvent:Connect(OnPlayerEnterShoot)
+    SHOOT_PERIMETER.endOverlapEvent:Connect(OnPlayerLeftShoot)
+
+    -- Loop through the activities table and add them to
+    -- the activity handler
+    for k, a in pairs(activities) do
+        ACTIVITY_HANDLER:AddActivity(k, activities[k])
+    end
+    ```
+
+## Create Damage Feedback
+
+In this section you will be creating a script to show the damage the player is doing to the boss. That damage will show up as numbers that will be visible only to the player applying the damage.
+
+### Create DamageClient Script
+
+1. Create a new script called **DamageClient**.
+
+    Place the **DamageClient** script into the **Client** group found under the **Boss Fight Map** group.
+
+2. Drag the **Fly Up Text Target** on to the **DamageClient** script as a custom property.
+
+    Name the custom property **FlyupPosition**.
+
+    The damage numbers that will show up need to know where to be positioned in the world. The **Fly Up Text Target** object inside the **Client** group found in the **Boss Geo** group will act as a helper object to get a reference to a position in the world.
+
+3. Add **Color** custom property.
+
+    Add a new **Color** custom property to the **DamageClient** script. Name the custom property `ImmuneColor`.
+
+    The color property will be used to change the color of the fly up text when the boss is immune to damage.
+
+4. Add the following code.
```

Comment:
More specific than code. It's fine to repeat what it is both in this title and in the explanation after.